app/views/bars/show.html.erb

```diff
@@ -40,24 +40,35 @@
     </div>
   </div>
 
-  <% @reviews.order(created_at: :desc).each do |review| %>
-    <div class="review">
-      <h3><%= review.title %></h3>
-      <h4><%= review.rating %>/5</h4>
-      <p><%= review.body %></p>
-    </div>
-
-    <% if user_signed_in? && (current_user == review.user || current_user.admin?) %>
-      <div class="button small">
-        <%= link_to "Delete Review", bar_review_path(@bar, review), data: { confirm: "Are you sure?" }, method: :delete %>
+<% @reviews.order(created_at: :desc).each do |review| %>
+  <div class="row">
+    <div id="review-<%= review.id %>" class="row review-container">
+      <div class="small-3 columns">
+        <%= link_to "<i id=\"upvote-review-#{review.id}\" class=\"fa fa-arrow-up\"><span class=\"sr-only\">Upvote Review #{review.id}</span></i>".html_safe, review_votes_path(review, value: 1), method: :post, class: "upvote", id: "upvote-#{review.id}" %>
+        <% if review.votes.nil? %>
+          <div class="votes" id="votes-review-<%= review.id %>">0</div>
+        <% else %>
+          <% total = review.votes.map {|v| v[:value]} %>
```

Comment:
Logic like this should be avoided in the view. I would put in a github issue to refactor this as a class method in the review model.